stl/inc/thread

```diff
@@ -270,9 +282,38 @@ _NODISCARD inline bool operator>=(thread::id _Left, thread::id _Right) noexcept
 
 _EXPORT_STD template <class _Ch, class _Tr>
 basic_ostream<_Ch, _Tr>& operator<<(basic_ostream<_Ch, _Tr>& _Str, thread::id _Id) {
-    return _Str << _Id._Id;
+    basic_ostringstream<_Ch, _Tr> _Out;
+    _Out.imbue(locale::classic());
+    _Out << _Id._Id;
+    return _Str << _Out.str();

```

Comment:
I guess that `to_string` is sufficient.

Reply:
Isn't `to_(w)string` affected by the current C locale?

Reply:
They are not... by default (they are when `_ENFORCE_FACET_SPECIALIZATIONS` is defined as non-zero). E.g. `basic_ostream<char32_t>` works by default despite not guaranteed by the standard.

IIUC we can still emulate `to_string` for other character types.

Reply:
> Isn't `to_(w)string` affected by the current C locale?

Only floating-point overloads are affected by C locale (which is changed by WG21-P2587R3 in C++26).

Reply:
After looking into the implementation of `num_put::do_put`, I think we can use `_UIntegral_to_buff` here:

https://github.com/microsoft/STL/blob/2261f7edb760eb3fe0726187c818b796dc7ea798/stl/inc/string#L441